sklearn/linear_model/base.py

```diff
@@ -436,14 +436,14 @@ def fit(self, X, y, sample_weight=None):
                 sample_weight).ndim > 1):
             sample_weight = column_or_1d(sample_weight, warn=True)
 
-        X, y, X_mean, y_mean, X_std = self._center_data(
-            X, y, self.fit_intercept, self.normalize, self.copy_X,
-            sample_weight=sample_weight)
-
         if sample_weight is not None:
             # Sample weight can be implemented via a simple rescaling.
             X, y = _rescale_data(X, y, sample_weight)
 
+        X, y, X_mean, y_mean, X_std = self._center_data(
+            X, y, self.fit_intercept, self.normalize, self.copy_X,
+            sample_weight=None)
```

Comment:
this is not equivalent as now you rescale the non-centered data. Which means that sample weight interact with intercept.

can you clarify why it's more correct this way?